### ISSUE TITLE
Calculate Position from another Position

### ISSUE THREAD

**Issue by yodalee** [pest-parser/pest#414]
I am writing a parser using Pest, and I use Position to store the token position.
I just got one question about the function `line_col` of Position.
I see the implementation and it will slice the input string, then iterate over it to find any newline character. The implementation is OK, but I wonder the performance will be worse as the position to the end of input.
I think it is possible that, given another Position, and line, col. Calculate the a new line, col based on the input.
If you think it is valuable to add, I think I can take the issue.

**Comment by CAD97** [comment]
Hmm, `line_col` does iterate the entire string.

I've done this for another library: ideally, it should rfind the last `\n` (e.g. with `memchr`) and then count `\n` bytes (e.g. with `bytecount`) to get the line number. This on its own can be a drastic speedup for late positions, with the cost of adding the two dependencies and maybe a little overhead for early positions.

Even so, it would be useful to offer the incremental version, for truly massive use cases. (Unless you need a massive number of line/col pairs, just working with byte positions and lazily normalizing to line/col for display is quite doable and very performant. A surprisingly big factor is how much space you save by saving approximately a `usize` per token.

TL;DR `line_col` can be trivially improved by `memchr::memrchr` and `bytecount::count`, and there's room for an incremental version as well.

**Comment by tomtau** [comment]
duplicate of https://github.com/pest-parser/pest/issues/560
